### PR TITLE
[feature-wip](multi-catalog) External file scan node

### PR DESCRIPTION
**Pull request by SaintBacchus**
# Proposed changes

Issue Number: close #xxx

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/planner/external/ExternalFileScanNode.java [hunk 1]
```diff
@@ -0,0 +1,432 @@
...
+    private static final Logger LOG = LogManager.getLogger(ExternalFileScanNode.class);
+
+    private static class ParamCreateContext {
+        public TBrokerScanRangeParams params;
+        public TupleDescriptor srcTupleDescriptor;
+        public Map<String, Expr> exprMap;
+        public Map<String, SlotDescriptor> slotDescByName;
+    }
+
+    private final org.apache.doris.catalog.Table catalogTable;
```
Comment (Jibing-Li) [inline fe/fe-core/src/main/java/org/apache/doris/planner/external/ExternalFileScanNode.java:140]: I think it's better to use the new table interface org.apache.doris.catalog.TableIf instead. But we can change it later as well.

Comment (SaintBacchus) [reply fe/fe-core/src/main/java/org/apache/doris/planner/external/ExternalFileScanNode.java:140]: Yes, now it only use it test the code of `FileScanNode`. When the `TableIf ` is ready, this pr will only support the external datasource.

File: fe/fe-core/src/main/java/org/apache/doris/analysis/TupleDescriptor.java [hunk 2]
```diff
@@ -352,7 +353,7 @@ public void getTableIdToColumnNames(Map<Long, Set<String>> tableIdToColumnNames)
             if (slotDescriptor.getColumn() != null) {
                 TupleDescriptor parent = slotDescriptor.getParent();
                 Preconditions.checkState(parent != null);
-                Table table = parent.getTable();
+                Table table = (Table) parent.getTable();
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/TupleDescriptor.java:356]: Should we use `TableIf` here?

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**